src/services/WaveformDataUtils.js

```diff
@@ -161,15 +165,55 @@ export default class WaveformDataUtils {
    * @param {Object} peaksInstance - current peaks instance for the waveform
    */
   activateSegment(id, peaksInstance) {
-    const segment = peaksInstance.segments.getSegment(id);
+    const validatedPeaks = this.initialSegmentValidation(id, peaksInstance);
```

Comment:
Does the result of initialSegmentValidation need to be stored?  It looks to me like that method is editing segments in place and not returning a new instance of peaksInstance.

Reply:
No it doesn't need to be stored. You are right! I didn't think of that. Thanks :+1: